packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -112,14 +146,14 @@ Widget _wrapWithBackground({
     );
   }
   final DecoratedBox childWithBackground = DecoratedBox(
-    decoration: BoxDecoration(
+    decoration: background != null ? background : BoxDecoration(
       border: border,
       color: backgroundColor,
     ),
     child: result,
   );
 
-  if (backgroundColor.alpha == 0xFF)
+  if (backgroundColor?.alpha == 0xFF)
     return childWithBackground;
 
   return ClipRect(
```

Comment:
You'd have to handle this case too. If you create a custom box decoration with a translucent background color, it should be blurred.